packages/nextjs/components/scaffold-eth/Balance.tsx

```diff
@@ -28,6 +28,10 @@ export const Balance = ({ address, className = "", usdMode }: BalanceProps) => {
 
   const [displayUsdMode, setDisplayUsdMode] = useState(price > 0 ? Boolean(usdMode) : false);
 
+  useEffect(() => {
+    setDisplayUsdMode(price > 0 ? Boolean(usdMode) : false);
+  }, [usdMode, price]);
```

Comment:
Also, in every `price` change `displayUsdMode` will change to `usdMode` (unexpected behavior/bug)

A bit more details

`usdMode` is `true`. `displayUsdMode` is `true`. Price is `3000`.  -> toggle `displayUsdMode` to `false` -> price changes to `3001` -> `useEffect` triggers. `displayUsdMode` changes to `true` because `usdMode` is `true`.

Reply:
This is currently not happening out of the box since enablePolling is set to false in `useNativeCurrencyPrice`, but it'll happen if we enable it.

I guess we could fix it with a ref? like `const prevPriceRef = useRef(price);` and only calling setDisplayUsdMode if `prevPriceRef.current` was 0 and now is > 0.

Reply:
Probably, let's fix it in another pr